power_grid_model_c/power_grid_model/include/power_grid_model/calculation_parameters.hpp

```diff
@@ -260,9 +260,14 @@ struct ComponentToMathCoupling {
 
 // change of update cause topology and param change, or just param change
 struct UpdateChange {
-    bool topo, param;
+    bool topo{};
+    bool param{};
 };
 
+constexpr UpdateChange operator||(UpdateChange const& a, UpdateChange const& b) {
```

Comment:
It is preferred to define `operator||` inside the class as a friend function. This will result a so-called hidden friend. It can be only access via ADL and will not participate the overload resolution in a random `||` operator. This increase compilation performance.

see:
https://youtu.be/iDX2d7poJnI?t=1553

Reply:
right, i'm so used to the `concepts` way of implementing this that i completely forgot about the advantages of using `friend` 😬 will fix